mountpoint-s3/tests/fuse_tests/rmdir_test.rs

```diff
@@ -42,6 +44,9 @@ where
 
     file.write_all(b"Hello World").unwrap();
     drop(file);
+    // when there are multiple tests called release method can be called later than unlink or rmdir.
+    // So, adding the sleep to ensure release being called earlier.
+    sleep(Duration::from_secs(1));
     let err =
         fs::remove_dir(&non_empty_dirpath).expect_err("removing non-empty directory should fail even after closing");
     assert_eq!(
```

Comment:
I think we'll be sad if we don't add a few retries to these sleeps, since sometimes the test runner might be very slow. Probably we want some kind of general function that retries a few times and then fails if none of the retries passed.

Reply:
Yes. I can add a macro which calls a methods  with a certain number of retries for it to succeed as was discussed yesterday.